concourse/scripts/kipper.sh

```diff
@@ -0,0 +1,361 @@
+#!/usr/bin/env bash
+
+# This script can be used in isolation, or as part of a concourse pipeline
+# to action fast shutdowns and startups of cf dev envs. All ec2 and rds 
+# instances tagged with the targetted env or created via the relevant rds
+# broker of that env are shutdown, except for those relating to bosh and
+# concourse to allow the env to start up again. The bosh reesurrector is
+# disabled to prevent bosh from trying to re-create the instances.
+# To use the script locally, you must first log in to the relevant bosh and
+# have the other pre-reqs installed.
+
+#TODO: docs, schedule, email/slack, pipeline?
+
+set -e -u -o pipefail
+shopt -s inherit_errexit
+
+main() {
+  parse_args "$@"
+  print_header
+  check_dependancies
+  validate_initial_state
+  if [[ "${action}" = "start" ]]; then
+    manage_db_instances_state
+    manage_ec2_instances_state
+    run_health_checks
+    toggle_bosh_resurrector "on"
+  fi
+  if [[ "${action}" = "stop" ]]; then
+    toggle_bosh_resurrector "off"
+    manage_ec2_instances_state
+    manage_db_instances_state
+  fi
+  echo "${env_arg} is now ${past_tense}"
+  echo "done."
+  echo
+}
+
+# action ec2 instances state change, loop waiting for target state
+manage_ec2_instances_state() {
```

Comment:
Where does it read the intended state from?

Reply:
the parse_args function sets some global variables for intended state...
  if [[ "${action_arg}" = 'kip' ]]; then
    action='stop'
    initial_ec2_state='running'
    target_ec2_state='stopped'
    initial_db_state='available'
    target_db_state='stopped'
    past_tense='abed'